### ISSUE TITLE
[BUG] Transport NPC's

### ISSUE THREAD

**Issue by cooler-SAI** [Arkania/ArkCORE-NG#114]
As youcan see Transport don't have NPC's
![wowscrnshot_032516_185307](https://cloud.githubusercontent.com/assets/812439/14047788/f88daf12-f2ba-11e5-9981-addcbf1d7919.jpg)

**Comment by graham79** [comment]
yeah, sadly that happens in a lot of places where transports leave from. Although if you are actually ON the transport you see the npc's where they should be. wierd.

**Comment by gpn39f** [comment]
what npc and quest is it??

**Comment by cooler-SAI** [comment]
it's not a quest - it's Zeppilins near Undercity

**Comment by gpn39f** [comment]
npc on Pictures, are the personal/passengers of Zeppelin?.. 

have you a map id of the Zeppelin???

**Comment by PiskelDEV** [comment]
I think they fall down from the Zeppelin when it leaves UC, and the NPCs will stay there while Zeppelin is away. Maybe a vmap/mmap issue?

**Comment by gpn39f** [comment]
I have found different places, where the crew of ships and zeppelin's standing around the landscape.
ships and airships as GameObject, using a own MapID. 
On our DB many npc's are using normal MapIds and not the ID from ship. so its easy to fix it: update the mapID for each ship, using the correct ShipMapID.. more difficulty is: compute the new relative position.. 

But One question: Why is it not same using gameobjects like chairs
When I enter the chairs in the same way in the gameobject, then the chairs are not visible in normal Map, and there are not visible on the ship to.
You can see them short moment on login, but then there are invisible to..
it's same in GM mode.

**Comment by gpn39f** [comment]
if i compare snif's then i see: Oure core send on opcode SMSG_UPDATE_OBJECT the flag: HasTransportData = false, blizz has here true.. ok...therefor the transport part is not send..,the posiition is part of the data.. 
has someone a fix for this??

**Comment by gpn39f** [comment]
in the SMSG_UPDATE_OBJECT i found 2 errors.
1.) the updateFlag missing a set GO_tansport_Flag(256) to true
2.) in the GameobjectTransportPart is the  GUID for the transport not send.

i have fixed both, the chairs are now visible on ships.. but now is there a failure more..
if using the chairs, and trying sitting down, you are teleported to relative position of map 0/1..
and not to map of the ship gameobject..
has someone a fix for this??
